expression/src/main/java/io/smallrye/common/expression/Expression.java

```diff
@@ -260,6 +260,9 @@ private static Node parseString(Itr itr, final boolean allowExpr, final boolean
                     }
                     // check to see if it's a dangling $
                     if (!itr.hasNext()) {
+                        if (flags.contains(Flag.NO_$$)) {
```

Comment:
What is the expected behavior here? IIRC this state is entered whenever we hit a `$` even if we've hit other things. So, I think we do not need to do anything special here because if `LENIENT_SYNTAX` is on, then we pass it through below and add the literal text from the start point to the end of the string. Otherwise it's an error. This is regardless of `NO_$$` AFAICT.

Reply:
Leftovers from the previous implementation.